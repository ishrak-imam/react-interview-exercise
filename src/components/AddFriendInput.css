:local(.addFriendInput) {
  border-radius: 0;
  border-color: #ABAAAA;
  border-left: 0;
  border-right: 0;
}

:local(.gender) {
  height: 120px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-start;
  background-color: white;
  padding: 10px
}

:local(.addFriend) {
  width: 100px
}
