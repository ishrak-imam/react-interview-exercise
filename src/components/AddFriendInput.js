import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import styles from './AddFriendInput.css'

class AddFriendInput extends Component {
  render () {
    const { name, gender } = this.state
    return (
      <div>
        <input
          type='text'
          autoFocus
          className={classnames('form-control', styles.addFriendInput)}
          placeholder='Type the name of a friend'
          value={name}
          onChange={this.handleChange}
          // onKeyDown={this.handleSubmit}
        />

        <div className={`${styles.gender}`}>
          <label>
            <input onChange={this.handleGenderChange} type='radio' value='M' checked={gender === 'M'} />
            <small className={`${styles.label}`}>Male</small>
          </label>

          <label>
            <input onChange={this.handleGenderChange} type='radio' value='F' checked={gender === 'F'} />
            <small className={`${styles.label}`}>Female</small>
          </label>

          <button
            onClick={this.handleSubmit}>
            <span>Add Friend</span>
          </button>
        </div>

      </div>
    )
  }

  constructor (props, context) {
    super(props, context)
    this.state = {
      name: this.props.name || '',
      gender: ''
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleGenderChange = this.handleGenderChange.bind(this)
  }

  handleGenderChange (e) {
    this.setState({ gender: e.target.value })
  }

  handleChange (e) {
    this.setState({ name: e.target.value.trim() })
  }

  handleSubmit (e) {
    // const name = e.target.value.trim()
    // const { gender } = this.state
    // if (e.which === 13) {
    //   this.props.addFriend({ name, gender })
    //   this.setState({ name: '', gender: '' })
    // }

    const { name, gender } = this.state
    this.props.addFriend({ name, gender })
    this.setState({ name: '', gender: '' })
  }
}

AddFriendInput.propTypes = {
  addFriend: PropTypes.func.isRequired
}

export default AddFriendInput
