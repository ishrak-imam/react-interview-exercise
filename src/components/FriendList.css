:local(.friendList) {
  padding-left: 0;
  margin-bottom: 0;
  margin-top: 10px
}
