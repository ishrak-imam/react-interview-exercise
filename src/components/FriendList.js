import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './FriendList.css'
import FriendListItem from './FriendListItem'

class FriendList extends Component {
  render () {
    const { friendIds } = this.props
    return (
      <ul className={styles.friendList}>
        {friendIds.map(id => {
          return (
            <FriendListItem key={id} friendId={id} />
          )
        })}
      </ul>
    )
  }
}

FriendList.propTypes = {
  friendIds: PropTypes.array.isRequired
}

export default FriendList
