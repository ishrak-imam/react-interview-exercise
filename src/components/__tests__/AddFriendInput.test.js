import React from 'react'
import { create } from 'react-test-renderer'
import AddFriendInput from '../AddFriendInput'
import FriendList from '../FriendList'

test('AddFriendInput component', () => {
  const component = create(
    <AddFriendInput addFriend={() => {}} />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
