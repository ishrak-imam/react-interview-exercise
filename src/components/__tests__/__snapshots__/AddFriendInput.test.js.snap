exports[`test AddFriendInput component 1`] = `
<div>
  <input
    autoFocus={true}
    className="form-control"
    onChange={[Function]}
    placeholder="Type the name of a friend"
    type="text"
    value="" />
  <div
    className="undefined">
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value="M" />
      <small
        className="undefined">
        Male
      </small>
    </label>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value="F" />
      <small
        className="undefined">
        Female
      </small>
    </label>
    <button
      onClick={[Function]}>
      <span>
        Add Friend
      </span>
    </button>
  </div>
</div>
`;
